operator/v1/types_console.go

```diff
@@ -298,6 +301,34 @@ type Perspective struct {
 	// visibility defines the state of perspective along with access review checks if needed for that perspective.
 	// +kubebuilder:validation:Required
 	Visibility PerspectiveVisibility `json:"visibility"`
+	// pinnedResources defines the list of default pinned resources that users will see on the perspective navigation if they have not customized these pinned resources themselves.
+	// The list of available Kubernetes resources could be read via `kubectl api-resources`.
+	// The console will also provide a configuration UI and a YAML snippet that will list the available resources that can be pinned to the navigation.
+	// Incorrect or unknown resources will be ignored.
+	// +kubebuilder:validation:MaxItems=100
+	// +optional
+	PinnedResources []PinnedResourceReference `json:"pinnedResources"`
+}
+
+// PinnedResourceReference includes the group, version and type of resource
+type PinnedResourceReference struct {
+	// Group is the API Group of the Resource.
+	// Enter empty string for the core group.
+	// Example: "", "apps", "build.openshift.io", etc.
+	// +kubebuilder:validation:Pattern:=^$|[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*
+	// +kubebuilder:validation:Required
+	Group string `json:"group"`
+	// Version is the API Version of the Resource.
+	// Example: "v1", "v1beta1", etc.
+	// +kubebuilder:validation:Pattern:=^[a-z0-9]+$
```

Comment:
We should add `// This value should consist of only lowercase alphanumeric characters.`